### PR TITLE
feat: add deleteContextsAsync and getAsyncProcess endpoints (delta-1534)

### PR DESCRIPTION
**Pull request by MihailTymoshenko**
[JIRA Ticket](https://bt.smartling.net/browse//DELTA-1534)

### CODE REVIEW

**Review by vbarhatov — APPROVED** [review]

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class AsyncProcessPTO implements ResponseData
+{
+    private String processUid;
+    private AsyncProcessState state;
```
Comment (vbarhatov) [inline smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java:15]: I've changed these parameters to `processState` and `processType`

File: smartling-contexts-api/src/test/java/com/smartling/api/contexts/v2/ContextsApiTest.java [hunk 2]
```diff
@@ -264,34 +282,40 @@ public void testMatchAsync() throws Exception
...
+        assertThat(asyncProcessPTO.getCreatedDate(), is("2017-04-19T14:21:11Z"));
+        assertThat(asyncProcessPTO.getModifiedDate(), is("2017-04-19T14:21:11Z"));
 
-        assertThat(matchStatusPTO.getMatchId(), is(MATCH_ID));
         RecordedRequest request = mockWebServer.takeRequest();
         assertEquals("GET", request.getMethod());
-        assertTrue(request.getPath().contains("/projects/" + PROJECT_UID + "/match/" + MATCH_ID));
-   }
+        assertTrue(request.getPath().contains("/projects/" + PROJECT_UID + "/processes/" + PROCESS_UID));
+    }
```
Comment (vbarhatov) [inline smartling-contexts-api/src/test/java/com/smartling/api/contexts/v2/ContextsApiTest.java:318]: extra space

**Review by vbarhatov — CHANGES_REQUESTED** [review]

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java [hunk 3]
```diff
@@ -63,10 +70,20 @@ MatchIdPTO matchAsync(@PathParam("projectId") String projectId, @PathParam("cont
...
 
+
+    /**
+     * @deprecated
+     * Use {@link com.smartling.api.contexts.v2.ContextsApi#getAsyncProcess(java.lang.String, java.lang.String)} instead
+     */
     @GET
     @Path("/projects/{projectId}/match/{matchId}")
+    @Deprecated
     MatchStatusPTO matchAsyncStatus(@PathParam("projectId") String projectId, @PathParam("matchId") String matchId);
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/ContextsApi.java]: done

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/MatchAsyncProcessResultPTO.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package com.smartling.api.contexts.v2.pto;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import java.util.List;
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class MatchAsyncProcessResultPTO implements AsyncProcessResultPTO
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/MatchAsyncProcessResultPTO.java:11]: in deserialising of response

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    private AsyncProcessType type;
+    private String createdDate;
+    private String modifiedDate;
+    @JsonDeserialize(contentAs = MatchAsyncProcessResultPTO.class)
+    private AsyncProcessResultPTO result;
```
[the remark this answers is not in the document]
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java:20]: Optional is not used in our endpoints and whole sdk

File: smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java (hunk 1 above)
Comment (MihailTymoshenko) [reply smartling-contexts-api/src/main/java/com/smartling/api/contexts/v2/pto/AsyncProcessPTO.java:15]: done

**Review by vbarhatov — APPROVED** [review]

**Pull request merged**